sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```
@@ -176,8 +174,12 @@ async def read(
         :returns: Dict representing the retrieved container.
         :rtype: Dict[str, Any]
         """
+        session_token = kwargs.pop('session_token', None)
```

I don't mind cleaning up docstrings and typed args in the method signature - I'm more concerned about changing the actual implementation behaviour to pretend these weren't passed in.
There are a lot of params in a lot of methods being changed here - can we confirm that every single one of these results in no behavioural changes?

---

I think if we've confirmed that the behaviour isn't changed (i.e. the service was either raising, silently ignoring them) - then I don't mind them being excluded.
If it's removed from docstrings, typehints, and not being sent to the service - for all intents and purposes in the async clients it's already removed with the added bonus of a warning. The only step beyond that would be to raise an error - which if they were being silently ignored before now, I would not do.

The only issue is likely in the synchronous clients in cases where they were also positional parameters - in which case, we cannot change the signature of a method without a major bump. If API/docs clean-up is the goal here - finding a way to add overloads so that those original positional parameters don't show up in the public surface is probably the easiest option. Alternatively you can do some other syntactic gymnastics on the method signature to support back-compat, but there will be a bit of overhead in maintaining that logic.

---

just went over the sync classes and it does seem as though thankfully all of these were keyword arguments to begin with based on the method definitions, so we should be good on that front